src/main/resources/com/google/api/codegen/nodejs/standalone_sample.snip

```diff
@@ -45,28 +44,212 @@
+      {@methodCallSampleCodeWithReturnValue(apiMethod, sample.sampleInitCode)}
     @else
-      {@methodCallSampleCodeWithoutReturnValue(apiMethod, sample.sampleInitCode, empty())}
+      {@methodCallSampleCodeWithoutReturnValue(apiMethod, sample.sampleInitCode)}
     @end
   @case "RequestAsyncPaged"
-    {@methodCallSampleCodeForPagedResponse(apiMethod, sample.sampleInitCode, empty())}
+    {@methodCallSampleCodeForPagedResponse(apiMethod, sample.sampleInitCode)}
   @case "RequestAsyncPagedAll"
-    {@methodCallSampleCodeForPagedResponseIterative(apiMethod, sample.sampleInitCode, empty())}
+    {@methodCallSampleCodeForPagedResponseIterative(apiMethod, sample.sampleInitCode)}
   @case "RequestStreamingBidi"
-    {@bidiStreamingSampleCode(apiMethod, sample.sampleInitCode, empty())}
+    {@bidiStreamingSampleCode(apiMethod, sample.sampleInitCode)}
   @case "RequestStreamingClient"
-    {@clientStreamingSampleCode(apiMethod, sample.sampleInitCode, empty())}
+    {@clientStreamingSampleCode(apiMethod, sample.sampleInitCode)}
   @case "RequestStreamingServer"
     {@initCode(apiMethod, sample.sampleInitCode)}
       {@methodCallSampleCode(apiMethod, sample.sampleInitCode)}.on('data', response => {
           // doThingsWith(response)
       });
   @case "LongRunningEventEmitter"
-    {@methodCallSampleCodeLongrunningEventEmitter(apiMethod, sample.sampleInitCode, empty())}
+    {@methodCallSampleCodeLongrunningEventEmitter(apiMethod, sample.sampleInitCode)}
   @case "LongRunningPromise"
-    {@methodCallSampleCodeLongrunningPromise(apiMethod, sample.sampleInitCode, empty())}
+    {@methodCallSampleCodeLongrunningPromise(apiMethod, sample.sampleInitCode)}
   @default
     $unhandledCallingForm: {@sample.callingForm} in sample "{@apiMethod.getClass.getSimpleName}"$
   @end
 @end
+
+@private methodCallSampleCodeWithReturnValue(apiMethod, init)
+  {@initCode(apiMethod, init)}
+  {@methodCallSampleCodePrefix(apiMethod, init)}
+    .then(responses => {
+      const response = responses[0];
+      // doThingsWith(response)
+    })
+    .catch(err => {
+      console.error(err);
+    });
+@end
+
+@private methodCallSampleCodeWithoutReturnValue(apiMethod, init)
+  {@initCode(apiMethod, init)}
+  {@methodCallSampleCodePrefix(apiMethod, init)}.catch(err => {
+    console.error(err);
+  });
+@end
+
+@private methodCallSampleCodeForPagedResponse(apiMethod, init)
+  // Or obtain the paged response.
+  @if initCode(apiMethod, init)
+    {@initCode(apiMethod, init)}
+
+  @end
+
+  const options = {autoPaginate: false};
+  const callback = responses => {
+    // The actual resources in a response.
+    const resources = responses[0];
+    // The next request if the response shows that there are more responses.
+    const nextRequest = responses[1];
+    // The actual response object, if necessary.
+    // const rawResponse = responses[2];
+    for (let i = 0; i < resources.length; i += 1) {
```

Comment:
`for (const resource of resources) { doThingsWith(resource); }` is more idiomatic